contracts/src/unruggable_memecoin.cairo

```diff
@@ -173,6 +195,19 @@ mod UnruggableMemecoin {
             amount: u256
         ) -> bool {
             let caller = get_caller_address();
+
```

Comment:
code duplication: both `transfer` and `transferFrom` should call an internal `_transfer` function that performs these checks, and then call `erc20._transfer` (available after rebasing on main)